fdbserver/BlobManifest.actor.cpp

```diff
@@ -343,15 +350,40 @@ class BlobManifestFileSplitter : public ReferenceCounted<BlobManifestFileSplitte
 	int64_t totalBytes_;
 	std::vector<Future<Void>> pendingFutures_;
 	bool closed_;
+	Optional<BlobGranuleCipherKeysCtx> cipherKeysCtx_;
 };
 
+ACTOR Future<BlobGranuleCipherKeysCtx> getLatestManifestCipherKeys(Reference<AsyncVar<ServerDBInfo> const> dbInfo,
+                                                                   Arena* arena) {
+	state BlobGranuleCipherKeysCtx cipherKeysCtx;
+	std::unordered_map<EncryptCipherDomainId, Reference<BlobCipherKey>> domainKeyMap =
+	    wait(GetEncryptCipherKeys<ServerDBInfo>::getLatestEncryptCipherKeys(
+	        dbInfo, { FDB_DEFAULT_ENCRYPT_DOMAIN_ID }, BlobCipherMetrics::BLOB_GRANULE));
+
+	auto domainKeyItr = domainKeyMap.find(FDB_DEFAULT_ENCRYPT_DOMAIN_ID);
+	ASSERT(domainKeyItr != domainKeyMap.end());
+	cipherKeysCtx.textCipherKey = BlobGranuleCipherKey::fromBlobCipherKey(domainKeyItr->second, *arena);
+
+	TextAndHeaderCipherKeys systemCipherKeys = wait(
```

Comment:
nit: `systemCipherKey` might not be misleading; we are choosing to use `default_domain` and not `fdb_system_keyspace_domain`; the reasoning being FDB system keyspace should be limited to data stored in FDB private system keyspace, manifest file stores the TenantBoundaries, hence, using `default_domain` seems more appropriate.

Reply:
fixed. thanks